clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -5038,7 +5034,7 @@ void SemaSYCL::CheckSYCLKernelCall(FunctionDecl *KernelFunc,
 
   // check that calling kernel conforms to spec
   QualType KernelParamTy = KernelFunc->getParamDecl(0)->getType();
-  if (not KernelParamTy->isReferenceType()) {
+  if (! KernelParamTy->isReferenceType()) {
     // passing by value.  emit warning if using SYCL 2020 or greater
     if (SemaRef.LangOpts.getSYCLVersion() >= LangOptions::SYCL_2020)
       Diag(KernelFunc->getLocation(), diag::warn_sycl_pass_by_value_deprecated);
```

Comment:
If we still have a diagnostic for deprecation for this case, you cannot assert in `GetSYCLKernelObjectType`